core/src/shred.rs

```diff
@@ -291,6 +315,22 @@ impl Shredder {
         first_shred
     }
 
+    fn new_coding_shred(
+        slot: u64,
+        index: u32,
+        num_data: usize,
+        num_code: usize,
+        position: usize,
+    ) -> CodingShred {
+        let mut coding_shred = CodingShred::default();
+        coding_shred.header.common_header.slot = slot;
+        coding_shred.header.common_header.index = index;
+        coding_shred.header.num_data_shreds = num_data as u8;
```

Comment:
Maybe we should bump up the `num_*` to u16s.

Reply:
done